apps/src/sites/studio/pages/cdoBlocklyWrapper.js

```diff
@@ -0,0 +1,108 @@
+  blocklyWrapper.wrapReadOnlyProperty('createSvgElement');
+  blocklyWrapper.wrapReadOnlyProperty('Css');
+  blocklyWrapper.wrapReadOnlyProperty('disableVariableEditing');
+  blocklyWrapper.wrapReadOnlyProperty('FieldAngleDropdown');
+  blocklyWrapper.wrapReadOnlyProperty('FieldAngleInput');
+  blocklyWrapper.wrapReadOnlyProperty('FieldAngleTextInput');
+  blocklyWrapper.wrapReadOnlyProperty('FieldButton');
+  blocklyWrapper.wrapReadOnlyProperty('FieldColour');
+  blocklyWrapper.wrapReadOnlyProperty('FieldColourDropdown');
+  blocklyWrapper.wrapReadOnlyProperty('FieldDropdown');
+  blocklyWrapper.wrapReadOnlyProperty('FieldIcon');
+  blocklyWrapper.wrapReadOnlyProperty('FieldImage');
+  blocklyWrapper.wrapReadOnlyProperty('FieldImageDropdown');
+  blocklyWrapper.wrapReadOnlyProperty('FieldLabel');
+  blocklyWrapper.wrapReadOnlyProperty('FieldParameter');
+  blocklyWrapper.wrapReadOnlyProperty('FieldRectangularDropdown');
+  blocklyWrapper.wrapReadOnlyProperty('FieldTextInput');
+  blocklyWrapper.wrapReadOnlyProperty('FieldVariable');
+  blocklyWrapper.wrapReadOnlyProperty('findEmptyContainerBlock');
+  blocklyWrapper.wrapReadOnlyProperty('fireUiEvent');
+  blocklyWrapper.wrapReadOnlyProperty('fish_locale');
+  blocklyWrapper.wrapReadOnlyProperty('Flyout');
+  blocklyWrapper.wrapReadOnlyProperty('FunctionalBlockUtils');
+  blocklyWrapper.wrapReadOnlyProperty('FunctionalTypeColors');
+  blocklyWrapper.wrapReadOnlyProperty('FunctionEditor');
+  blocklyWrapper.wrapReadOnlyProperty('functionEditor');
+  blocklyWrapper.wrapReadOnlyProperty('gamelab_locale');
+  blocklyWrapper.wrapReadOnlyProperty('Generator');
+  blocklyWrapper.wrapReadOnlyProperty('getRelativeXY');
+  blocklyWrapper.wrapReadOnlyProperty('googlecode');
+  blocklyWrapper.wrapReadOnlyProperty('hasCategories');
+  blocklyWrapper.wrapReadOnlyProperty('html');
+  blocklyWrapper.wrapReadOnlyProperty('inject');
+  blocklyWrapper.wrapReadOnlyProperty('INPUT_VALUE');
+  blocklyWrapper.wrapReadOnlyProperty('js');
+  blocklyWrapper.wrapReadOnlyProperty('mainBlockSpace');
+  blocklyWrapper.wrapReadOnlyProperty('mainBlockSpaceEditor');
+  blocklyWrapper.wrapReadOnlyProperty('modalBlockSpace');
+  blocklyWrapper.wrapReadOnlyProperty('Msg');
+  blocklyWrapper.wrapReadOnlyProperty('Names');
+  blocklyWrapper.wrapReadOnlyProperty('netsim_locale');
+  blocklyWrapper.wrapReadOnlyProperty('Procedures');
+  blocklyWrapper.wrapReadOnlyProperty('removeChangeListener');
+  blocklyWrapper.wrapReadOnlyProperty('RTL');
+  blocklyWrapper.wrapReadOnlyProperty('SVG_NS');
+  blocklyWrapper.wrapReadOnlyProperty('tutorialExplorer_locale');
+  blocklyWrapper.wrapReadOnlyProperty('useContractEditor');
+  blocklyWrapper.wrapReadOnlyProperty('useModalFunctionEditor');
+  blocklyWrapper.wrapReadOnlyProperty('Variables');
+  blocklyWrapper.wrapReadOnlyProperty('weblab_locale');
+  blocklyWrapper.wrapReadOnlyProperty('Xml');
+
+  blocklyWrapper.wrapSettableProperty('assetUrl');
```

Comment:
[out of curiosity] how did you know which properties should be readonly vs read-write? did you just check the usage in our codebase?

Reply:
There definitely might've been a better way to do this haha, but I grepped for "Blockly." and "blockly." in the whole codebase, put that into [this spreadsheet](https://docs.google.com/spreadsheets/d/17c0VRX9GV9SKTxbfKPoPmlSLWT5UoWzsGroSts3wuzE/edit#gid=1456923540), then filtered for rows that contained `=` to find the properties that we set (captured in the Setting Properties tab of that sheet)

Reply:
lol sometimes that's the best way to do it